server_modules/file_system.js

```diff
@@ -0,0 +1,134 @@
+var fs = require('fs');
+var Q = require('q');
+var mkdirp = require('mkdirp');
+var path = require('path')
+var utils = require('./utils');
+var args = require('./args');
+
+function absolutePath(file) {
+    return path.resolve(path.dirname(process.argv[1]), file);
+}
+
+function getDataFilePath(file) {
+    return path.resolve(path.dirname(process.argv[1]), args.datadir, file);
+}
+
+function parseFile(data) {
+    return Q.promise(function(resolve,reject) {
+        try {
+            var res = JSON.parse(data);
+            resolve(res);
+        } catch(e) {
+            reject(e);
+        }
+    });
+}
+
+exports.resolve = function(file) {
+    return path.resolve(absolutePath(file));
+};
+
+exports.readFile = function(file) {
+    file = absolutePath(file);
+
+    return Q.promise(function(resolve,reject) {
+        fs.exists(file,function(exists) {
+            if (exists) {
+                resolve(exists);
+            } else {
+                reject({
+                    status: 404,
+                    message: 'file not found'
+                });
+            }
+        });
+    }).then(function() {
+        return Q.nfcall(fs.readFile, file, "utf-8").catch(function(e) {
+            throw new Error({
+                status: 500,
+                message: 'error reading file'
+            });
+        });
+    });
+}
+
+exports.parseDataFile = function(file) {
+    return exports.parseFile(getDataFilePath(file));
```

Comment:
`exports` doesn't have `parseFile()` (it's a local function).
Additionally, `parseFile()` expects the file contents, not a filename (I agree the name is confusing, change it if you like).